### ISSUE TITLE
IF5181 Pengenalan Pola

### ISSUE THREAD

**Issue by banditelol** [banditelol/SistemIntelijenS2ITB#4]
# Pengenalan Pola

**Comment by banditelol** [comment]
## 2020-12-02 Presentasi Tubes

### Jaffe Facial Expression
Gisel dan Rossevine

![image](https://user-images.githubusercontent.com/5263688/100852283-dd846400-34b8-11eb-8a2e-fdeae436e4c1.png)

> Question
> Kenapa kita splitting datanya 60:20:20 ? Ga 70:15:15
> Kenapa kita make AP aja? kenapa ga make F1 score?

#### Pertanyaan Bu Masayu

Gisela & Rossevina 16.03-16.13 (lebih 3 menit)
1. Desain experimen apakah dipengaruhi oleh studi literatur ?
2. Pada grid-search, RF 5-fold cv tapi XGBoost 10-fold cv ?
3. Image preprocessing pada CNN dilakukan sebelum conv-layer pertama ?
4. Activation function knp ada yg pakai sigmoid dan softmax ? Tasknya klasifikasi multiclass ?
5. Apakah hasil yg ditampilkan baru menggunakan data validasi saja ? Mana hasil dari data uji ?

1. Kemarin sempat melihat beberapa referensi Bu, untuk perubahan jumlah filter jg ada studi litelatur lain yang melakukan, namun disini kami menggunakan jumlah filter yang berbeda, karena ketika menggunakan sesuai dengan studi literatur tersebut hasil yang diperoleh kurang optimal, untuk image processing juga dari beberapa studi Bu, dan kami tertarik untuk membandingkannya.
2. Iya Bu, kami melakukan pembedaan dalam grid search, untuk model random forest digunakan kfold dengan fold 5, kemudian XGB kami mencoba menggunakan StratifiedKFold dengan fold 10.
3. Iya diawal gambar di crop seukuran wajah, kemudian dilakukan preprocessing dengan LBP, dan HE-IN, datanya kami simpan. Kemudian kami load kembali untuk dipisahkan berdasarkan rasio 70:15:15, baru kemudian data train dan val digunakan untuk membangun model cnn.
4. Klasifikasi multi class Bu, sebelumnya kami menggunakan softmax, namun hasilnya dirasa kurang optimal, ketika membaca beberapa penjelasan mengenai klasifikasi gambar, ada beberapa yang menyarankan activation function sigmoid, karena hasilnya lebih optimal dari softmax, sehingga kami ingin membandingkan mana activation function yang lebih baik.

### Online Shopper
Rifqi dan Vicky

![image](https://user-images.githubusercontent.com/5263688/100854523-b1b6ad80-34bb-11eb-868a-3e31b1bfdc01.png)

Mereka menggunakan referensi splitting dari paper.

> Renungan
> Kita menggunakan split dari valdiasi tapi ga ada referensi karena kebanyakan make COCO secara penuh.

![image](https://user-images.githubusercontent.com/5263688/100855646-fc84f500-34bc-11eb-9136-34457a1cca6e.png)

#### Pertanyaan

1. Ref Kabir dan Okan akurasi mirip, tapi F1 jauh berbeda, apakah ada penjelasannya ?
2. Variabel response itu ukuran kinerja (F1 macro), bukan algoritma dgn parameter.
3. Kenapa harus membatasi jumlah daun pada eksperimen decision tree ?
4. Jika ada potensi perbedaan cost false classification, kenapa tidak menggunakan model berbasis cost ?

### Credit Card Fraud
Fajar dan Thariq

> Renungan
> Bikin juga EDA soal berapa banyak data yang ga ada anotasinya dan apa yang kita lakuin

#### Pertanyaan

Fajar & Marsa 16.41-16.51
sangat imbalanced 99% non-fraud
1. Lengkapi desain eksperimen. Sebelumnya terdapat EDA dan seleksi fitur, masukkan sbg bagian dari desain eksperimen.
2. PCA 27 lebih baik. angka 27 ini dari mana ? 
3. Kesimpulan 1 & 2 saling berkaitan.
4. Apakah XGBoost overfit dilihat dari rekap pengujian ?

### Text Data
Rifo dan Dandy

Menggunakan NN dengan arsitektur sendiri

![image](https://user-images.githubusercontent.com/5263688/100858517-ac0f9680-34c0-11eb-8b5f-4588bc40079a.png)

> Question
> Untuk object detection bisa ga sih kita ngelakuin error analysis gini ? ada interface yang mudah g untuk ngelakuin ini?

![image](https://user-images.githubusercontent.com/5263688/100858593-c47fb100-34c0-11eb-9ae7-79047777d9d9.png)

![image](https://user-images.githubusercontent.com/5263688/100858958-3a841800-34c1-11eb-8cfc-a64d07de71bb.png)

#### Pertanyaan

1. Kebanyakan literatur menggunakan transfer learning dgn LM atau LSTM. Kenapa malah ANN yg digunakan sangat sederhana, dan tidak mempertimbangkan teks sbg sekuens kata ? walaupun error rate model terbaik 9 didapatkan dari ANN.
2. Fitur (representasi teks) apa yg digunakan dalam eksperimen DTL, RF, ataupun XGBoost ?
3. Analisis hasilnya sudah bagus. Apakah bisa diduga apa yg menyebabkan ANN memberi error rate terendah ?

**Comment by banditelol** [comment]
## 2010-12-04 Presentasi Tubes

### Fairuz & Gabriel (Spambase) 15.00-15.10
1. Makalah 1-3 menggunakan skema validasi yang berbeda, apakah akurasinya bisa dibandingkan ?
2. Analisis hasil untuk data yang masih salah diklasifikasi perlu dilakukan untuk mencari penyebab kesalahan.
3. 27 dari 175 konfigurasi SVM dgn akurasi di atas 90%. Apa yg ingin disampaikan dari heatmap yg diberikan ?
Silahkan dijawab di gclassroom ya

### Made & Nyoman 15.23-15.33
1. Seperti apa lingkungan eksperimen yang dilakukan ?
2. Video tesnya diambil berapa frame ? model yg digunakan yg mana ? dan kinerjanya berapa ?

[3:24 PM, 12/4/2020] Aditya Rachman Putra: intinya mau nunjukkin soal kita make validasi buat ngambil model terbaik, terus make test buat ngeliat performa modelnya
[3:25 PM, 12/4/2020] Aditya Rachman Putra: dan kalau error test-val terlalu beda errornya berarti kita terlalu overfit ke validasi
[3:25 PM, 12/4/2020] Harits: hoo woke
[3:25 PM, 12/4/2020] Harits: shit 40k
[3:25 PM, 12/4/2020] Aditya Rachman Putra: damn
[3:25 PM, 12/4/2020] Aditya Rachman Putra: hahaha
[3:25 PM, 12/4/2020] Aditya Rachman Putra: made kuat PCnya ceunah
[3:26 PM, 12/4/2020] Harits: hmmmmm kalo 2080 TI ga cukup juga sih dit wkwk
[3:26 PM, 12/4/2020] Harits: RAMnya di bawah kernel GPU kaggle
[3:26 PM, 12/4/2020] Aditya Rachman Putra: mungkin K80 x 4 wkwkwk
[3:26 PM, 12/4/2020] Harits: unless ada cluster GPU
[3:26 PM, 12/4/2020] Harits: aih berasa asing ada produk gitu di luar industri/kampus
[3:26 PM, 12/4/2020] Aditya Rachman Putra: wkwkwk
[3:27 PM, 12/4/2020] Aditya Rachman Putra: terus baru tau kl nyoman mbahas CSP
[3:27 PM, 12/4/2020] Aditya Rachman Putra: btw kita urutan berapa sih?
[3:27 PM, 12/4/2020] Aditya Rachman Putra: urang lupa nyimpen
[3:27 PM, 12/4/2020] Aditya Rachman Putra: hahaha
[3:27 PM, 12/4/2020] Harits: asking the same question
[3:28 PM, 12/4/2020] Harits: btw urang masih bingung cara kerja anchor, kerasa lebih make sense yang anchor free
[3:28 PM, 12/4/2020] Aditya Rachman Putra: hahahaa
[3:28 PM, 12/4/2020] Harits: kalo errornya ga jauh berarti variancenya kurang?
[3:28 PM, 12/4/2020] Harits: low variance bukan?
[3:28 PM, 12/4/2020] Harits: eh ga ngaruh ya
[3:29 PM, 12/4/2020] Aditya Rachman Putra: ga ngaruh pak hahaha, lagian variance teh buat ngukur beda val-train error
[3:29 PM, 12/4/2020] Harits: hoo, bukan ke val-test ya. Woke noted
[3:29 PM, 12/4/2020] Aditya Rachman Putra: iyoo
[3:29 PM, 12/4/2020] Aditya Rachman Putra: Made terdengar udah sangat pengalaman make NN
[3:30 PM, 12/4/2020] Harits: aing mikir, pytorch itu pretrainednya di imagenet atau coco
[3:30 PM, 12/4/2020] Aditya Rachman Putra: imagenet pak
[3:30 PM, 12/4/2020] Aditya Rachman Putra: udah urang masukin di slide kl ga salah
[3:31 PM, 12/4/2020] Aditya Rachman Putra: jadi kepikiran kita image sizenya kegedean kali ya hahaha
[3:31 PM, 12/4/2020] Harits: woh woke
[3:33 PM, 12/4/2020] Aditya Rachman Putra: gils FPSnya bs tinggi gitu
[3:33 PM, 12/4/2020] Harits: iya urang penasaran gimana caranya
[3:33 PM, 12/4/2020] Harits: aktivasi yang mish juga baru denger
[3:33 PM, 12/4/2020] Aditya Rachman Putra: hmmm iyaya urang kenapa ga plot error categoricalnya ya?
[3:34 PM, 12/4/2020] Aditya Rachman Putra: 10) 23520011 - 23520039
11) 23520008 - 23520901
12) 23520047 - 23520032
13) 23520050 - 23520045

masih 3 kelompok lagi
[3:34 PM, 12/4/2020] Aditya Rachman Putra: duh urang iri deh
[3:34 PM, 12/4/2020] Aditya Rachman Putra: pengen ngerun di video juga wkwkkw
[3:34 PM, 12/4/2020] Harits: hmmmm urang udah kepikiran ribetnya gimana
[3:35 PM, 12/4/2020] Harits: wkwkwk ngga ada fasilitasnya pak
[3:35 PM, 12/4/2020] Aditya Rachman Putra: nitip run di made wkwk
[3:35 PM, 12/4/2020] Harits: apa karena yolo lebih murah dari cost komputasinya ya?
[3:35 PM, 12/4/2020] Harits: tanya Made wkwk
[3:35 PM, 12/4/2020] Aditya Rachman Putra: harusnya g bgt ribet sih, soalnya si codenya dia udah outputin error di masing2 step, tinggal di log aja
[3:36 PM, 12/4/2020] Aditya Rachman Putra: urang sepertinya akan ngajak ngobrol made dalam waktu dekat wkwk
[3:36 PM, 12/4/2020] Harits: wkwk urang ikut dit

### Irfan & Adi (CIFAR-10) 15.34-15.44 
1. Seperti apa lingkungan eksperimen yang dilakukan ?
2. Model ResNetXX digunakan untuk feature extraction. Apakah pernah dilakukan pemodelan tanpa RestNetXX ? CNN harusnya bisa digunakan langsung ke raw data.
3. Apa dasar jumlah epoch membuat kita menyimpulkan apakah modelnya underfit atau overfit ?

### Nadya & Fauzan (Oxford17Flower) 15.45-15.55 => lebih 5 menit
1. Kenapa masih menggunakan descriptor utk skenario DTL, bukan langsung CNN seperti referensi ?
2. Apakah ada dugaan mengapa XGBoost memberi akurasi tes terbaik dibanding CNN atau Inception ?
3. Bgm mengetahui modelnya cenderung underfit ? Bisa berikan referensi terkait penjelasan pada slide terakhir
